applications/FluidDynamicsApplication/custom_processes/calulate_levelset_consistent_nodal_gradient_process.h

```diff
@@ -0,0 +1,183 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Mohammad R. Hashemi
+//
+
+#ifndef KRATOS_LEVELSET_CONSISTENT_NODAL_GRADIENT_H
+#define KRATOS_LEVELSET_CONSISTENT_NODAL_GRADIENT_H
+
+// Project includes
+#include "processes/process.h"
+
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Utility to due the redistancing based on the time-dependednt Eikonal equation
```

Comment:
I think this documentation if from other process.

Reply:
Would be nice to add some info about this too. Explaining that we do the nodal projection taking into account the position of the level set (intersected elements are not considered for this).

Reply:
Done.